src/main/java/de/tum/cit/aet/artemis/exercise/service/ExerciseService.java

```diff
@@ -346,18 +346,19 @@ public Set<Exercise> loadExercisesWithInformationForDashboard(Set<Long> exercise
      * Filter all exercises for a given course based on the user role and course settings
      * Assumes that the exercises are already been loaded (i.e. no proxy)
      *
-     * @param course corresponding course: exercises
-     * @param user   the user entity
+     * @param course                      corresponding course: exercises
+     * @param user                        the user entity
+     * @param reloadOnlineCourseExercises this is only necessary when fetching a single course
```

Comment:
Okay, but why do we need to reload them?

Reply:
this is a temporary fix, we should completely separate the code between loading multiple courses and one course in the future.